source/client_channel_handler.c

```diff
@@ -160,10 +136,37 @@ static int s_packet_handler_connack(
             "id=%p: invalid connect return code %d, disconnecting",
             (void *)connection,
             connack.connect_return_code);
-        /* If error code returned, disconnect */
+        /* If error code returned, disconnect, on_completed will be invoked from shutdown process */
```

Comment:
Can you link where this is done?

Reply:
https://github.com/awslabs/aws-c-mqtt/blob/master/source/client.c#L181
The return code will not be shown to the client. Only the error code `AWS_ERROR_MQTT_PROTOCOL_ERROR` will be there

Reply:
Nice.